From e442a9c8fe574fe7f4a2bcd3a4e1980b79b26fb9 Mon Sep 17 00:00:00 2001
From: Jiangyi <sam.andrew.jiang@gmail.com>
Date: Fri, 18 Jul 2014 10:50:52 -0400
Subject: [PATCH] frameworks/webview: Use prebuilt java library for building if
 using prebuilt Chromium

webviewchromium requires android_java_webview for building, but if prebuilt Chromium is
used, android_java_webview does not get built. Therefore, in this case, get webviewchromium
to rely on a prebuilt android_java_webview jar in order to not break build.

Change-Id: I60f8f0fc7d63eb16b90ddce03690782b1b2f3026
(cherry picked from commit 3d6c233caf5177fe7b11237b844301733b323c4d)

Conflicts:
	chromium/Android.mk
---
 chromium/Android.mk | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/chromium/Android.mk b/chromium/Android.mk
index edcfdb2..a0c933e 100644
--- a/chromium/Android.mk
+++ b/chromium/Android.mk
@@ -19,6 +19,13 @@
 LOCAL_PATH := $(call my-dir)
 CHROMIUM_PATH := external/chromium_org
 
+# Use prebuilt android_webview_java to satisfy webviewchromium's dependency
+# if we are working with prebuilt Chromium
+ifeq ($(PRODUCT_PREBUILT_WEBVIEWCHROMIUM),yes)
+include $(CLEAR_VARS)
+LOCAL_PREBUILT_STATIC_JAVA_LIBRARIES := android_webview_java:../../../prebuilts/chromium/$(TARGET_DEVICE)/android_webview_java.jar
+include $(BUILD_MULTI_PREBUILT)
+endif
 # Native support library (libwebviewchromium_plat_support.so) - does NOT link
 # any native chromium code.
 include $(CLEAR_VARS)
-- 
2.1.4

